app/controllers/teams/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Teams::UsersController < AuthenticatedController
+   before_action -> { authorize team, policy_class: TeamUsersPolicy }
+
+    def remove_confirm
+      @team = team
+      @user = team.users.find_by(id: params[:id])
+      if @user.present? && @user != current_user
+        render :remove_confirm      
+      else
+        render_401
+      end
+    end
+  
+    def destroy
+      user = team.users.find_by(id: params[:id])
+      if user.present? && user != current_user
+        team.users.delete(user)
+        flash[:success] = I18n.t('team.users.remove_success', email: user.email)
+        redirect_to team_path(@team.id)
```

Comment:
You can change the reference to team here as just "team" with no "@", since the team method below is taking care of "memoizing" (i.e. caching) the team.